core/src/main/scala/sttp/tapir/Codec.scala

```diff
@@ -585,15 +610,16 @@ case class MultipartCodec[T](rawBodyType: RawBodyType.MultipartBody, codec: Code
 }
 
 object MultipartCodec extends MultipartCodecMacros {
-  val Default: MultipartCodec[Seq[Part[Array[Byte]]]] = {
+  private val arrayBytePartListCodec = implicitly[Codec[List[Part[Array[Byte]]], List[Part[Array[Byte]]], OctetStream]]
+
+  val Default: MultipartCodec[Seq[Part[Array[Byte]]]] =
     Codec
-      .multipart(Map.empty, Some(PartCodec(RawBodyType.ByteArrayBody, Codec.listHead(Codec.byteArray))))
-      .asInstanceOf[MultipartCodec[Seq[Part[Array[Byte]]]]] // we know that all parts will end up as byte arrays
-  }
+      .multipart(Map.empty, Some(PartCodec(RawBodyType.ByteArrayBody, arrayBytePartListCodec)))
+      // we know that all parts will end up as byte arrays; also, removing the by-name grouping of parts
+      .map(_.values.toSeq.flatMap(_.asInstanceOf[List[Part[Array[Byte]]]]))(l => ListMap(l.map(p => p.name -> p): _*))
```

Comment:
@micossow I simplified the default codec here, let me know what you think. Tests seem to pass ;-)